GDevelop.js/TestUtils/CodeGenerationHelpers.js

```diff
@@ -309,6 +334,70 @@ function generateCompiledEventsFromSerializedEvents(
   return runCompiledEvents;
 }
 
+/**
+ * Helper to create compiled events from serialized events, creating a project and the events function.
+ * @param {*} gd
+ * @param {gdEventsFunctionExtension} extension
+ * @param {gdSerializerElement} eventsSerializerElement
+ * @param {{parameterTypes: {[name: string]: string}, groups: {[name: string]: string[]}, logCode: boolean}?} configuration
+ * @returns
+ */
+function generateCompiledEventsFunctionFromSerializedEvents(
+  gd,
+  extension,
+  eventsSerializerElement,
+  configuration
+) {
+  const project = new gd.ProjectHelper.createNewGDJSProject();
+  const eventsFunction = new gd.EventsFunction();
+  eventsFunction.getEvents().unserializeFrom(project, eventsSerializerElement);
+
+  if (configuration) {
+    const { parameterTypes, groups } = configuration;
+    if (groups) {
+      for (const groupName in groups) {
+        if (Object.hasOwnProperty.call(groups, groupName)) {
```

Comment:
```suggestion
        if (groups.hasOwnProperty(groupName)) {
```

should work. But not sure you even need this? I think you can remove it.